agent-service/agents/shared/rag/rag_client.py

```
@@ -0,0 +1,79 @@
+
+async def run(
+    message: str,
+    *,
+    system_prompt: str,
+    config: Dict[str, Any],
+    logger,
+) -> Dict[str, Any]:
+    """Run retrieval + LLM generation with a shared prompt template."""
+    try:
+        llm = LLMProvider()
+    except Exception as exc:
```

Ruff (0.14.14)

Do not catch blind exception: `Exception`

(BLE001)